app/providers/atmosphere/optimization_strategy/default.rb

```diff
@@ -106,6 +111,21 @@ def select
           tmpls.each do |tmpl|
             candidate_tenants = get_candidate_tenants_for_template(tmpl)
             candidate_tenants.each do |t|
+              # The next step is to restrict tenants by user funds.
+              # A tenant is only valid for use if it
+              # shares a fund with the appliance's owner. Additionally,
+              # if the appliance fund is explicitly specified
+              # in the instantiation requests, the selection must be honored.
+              unless @appliance.blank?
```

Comment:
Yeah, pretty much - it's used to determine how much a new appliance *would* cost (needed by UI) so it gets invoked without an appliance argument.